lib/tasks/plugin.rake

```diff
@@ -290,3 +290,124 @@ task "plugin:versions" do |t, args|
 
   puts JSON.pretty_generate(versions)
 end
+
+desc "create a new plugin based on template"
+task "plugin:create", [:name] do |t, args|
+  class StringHelpers
+    def self.to_snake_case(string)
+      string.dup.gsub!("-", "_")
+    end
+
+    def self.to_pascal_case(string)
+      string.dup.split("-").map(&:capitalize).join
+    end
+
+    def self.to_pascal_spaced_case(string)
+      string.dup.split("-").map(&:capitalize).join(" ")
+    end
+
+    def self.is_in_kebab_case?(string)
+      string.match?(/\A[a-z0-9]+(-[a-z0-9]+)*\z/)
+    end
+  end
+
+  plugin_name = args[:name]
+
+  abort("Supply a name for the plugin") if plugin_name.blank?
+  abort("Name must be in kebab-case") unless StringHelpers.is_in_kebab_case?(plugin_name)
+
+  plugin_path = File.expand_path("plugins/" + plugin_name)
+
+  abort("Plugin directory, " + plugin_path + ", already exists.") if File.directory?(plugin_path)
+
+  promises = []
+  failures = []
+  repo = "https://github.com/discourse/discourse-plugin-skeleton"
+  promises << Concurrent::Promise.execute do
```

Comment:
Do we really need `Concurrent::Promise` here? Seems like we're only running one thing at a time, so it could just be a `begin/end` block with retry?